src/main/java/com/uber/cadence/converter/JsonDataConverter.java

```diff
@@ -88,6 +90,11 @@ public JsonDataConverter(Function<GsonBuilder, GsonBuilder> builderInterceptor)
     try {
       if (values.length == 1) {
         Object value = values[0];
+        if (value instanceof GeneratedMessageV3) {
+          GeneratedMessageV3 protoObj = (GeneratedMessageV3) value;
+          return protoObj.toByteArray();
```

Comment:
I think by default we want to serialize it to JSON to make it UI friendly: https://github.com/protocolbuffers/protobuf/blob/master/java/util/src/main/java/com/google/protobuf/util/JsonFormat.java